app/src/main/java/nu/yona/app/api/manager/impl/NotificationManagerImpl.java

```diff
@@ -161,6 +162,87 @@ public void onError(Object errorMessage) {
         }
     }
 
+
+    public void getMessageWithUrl(final String urlForMessagesFetch, final boolean isUnreadStatus, final DataLoadListener listener, final boolean isProcessUpdate) {
+        try {
+            User user = YonaApplication.getEventChangeManager().getDataState().getUser();
+            if (user != null && user.getLinks() != null && user.getLinks().getYonaMessages() != null
+                    && !TextUtils.isEmpty(user.getLinks().getYonaMessages().getHref())) {
+
+                notificationNetwork.getNextSetOfMessagesFromURL(urlForMessagesFetch, YonaApplication.getEventChangeManager().getSharedPreference().getYonaPassword(), isUnreadStatus, new DataLoadListener() {
+                    @Override
+                    public void onDataLoad(Object result) {
+                        if (listener != null) {
```

Comment:
Can this ``listener`` realistically be null?

Reply:
Yes. it can be null. For eg: Please see line no:226 in LoginActivity class, where we dont listen for the call back         APIManager.getInstance().getActivityCategoryManager().getActivityCategoriesById(null);

Reply:
OK, clear.